webapp/channels/src/components/gif_picker/gif_picker.tsx

```diff
@@ -1,90 +1,38 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
-import React, {useState} from 'react';
+import React, {SyntheticEvent} from 'react';
+import {IGif} from '@giphy/js-types';
 
-import {GifsAppState, GfycatAPIItem} from '@mattermost/types/gifs';
-
-import App from 'components/gif_picker/components/App';
-import Categories from 'components/gif_picker/components/Categories';
-import Search from 'components/gif_picker/components/Search';
-import Trending from 'components/gif_picker/components/Trending';
-import constants from 'components/gif_picker/utils/constants';
-
-export const appProps: GifsAppState = {
-    appName: constants.appName.mattermost,
-    basePath: '/mattermost',
-    itemTapType: constants.ItemTapAction.SHARE,
-    appClassName: 'gfycat',
-    shareEvent: 'shareMattermost',
-    appId: 'mattermostwebviews',
-    enableHistory: true,
-    header: {
-        tabs: [constants.Tab.TRENDING, constants.Tab.REACTIONS],
-        displayText: false,
-    },
-};
+import GifPickerSearch from './components/gif_picker_search';
+import GifPickerItems from './components/gif_picker_items';
 
 type Props = {
+    filter: string;
     onGifClick?: (gif: string) => void;
-    defaultSearchText?: string;
-    handleSearchTextChange: (text: string) => void;
+    handleFilterChange: (filter: string) => void;
 }
 
 const GifPicker = (props: Props) => {
-    const [action, setAction] = useState(props.defaultSearchText ? 'search' : 'trending');
+    const handleItemClick = (gif: IGif, event: SyntheticEvent<HTMLElement, Event>) => {
+        if (props.onGifClick) {
+            event.preventDefault();
 
-    const handleTrending = () => setAction('trending');
-    const handleCategories = () => setAction('reactions');
-    const handleSearch = () => setAction('search');
-
-    const handleItemClick = (gif: GfycatAPIItem) => {
-        props.onGifClick?.('![' + gif.title?.replace(/]|\[/g, '\\$&') + '](' + gif.max5mbGif + ')');
+            const imageWithMarkdown = `![${gif.title}](${gif.images.fixed_height_downsampled.url})`;
```

Comment:
Did you have a chance to confirm with UX about which of these images they'd prefer us to use? I assume this one is probably good though

Reply:
@abhijit-singh here are the following sample image data we receive from the API, the one I currently have is `fixed_height_downsampled`. But please let me know from the following if there are other preferences, It contains height, width, size

```
{
    "original": {
        "height": 480,
        "width": 480,
        "size": "5513844",
        "url": "https://media1.giphy.com/media/dSTZlrbH8u5jI6Hd51/giphy.gif?cid=34cf01dco5jtrtfj8psd7ykpeppfup8d6p8xy1qwssfsmise&ep=v1_gifs_trending&rid=giphy.gif&ct=g",
        "mp4_size": "639713",
        "mp4": "https://media1.giphy.com/media/dSTZlrbH8u5jI6Hd51/giphy.mp4?cid=34cf01dco5jtrtfj8psd7ykpeppfup8d6p8xy1qwssfsmise&ep=v1_gifs_trending&rid=giphy.mp4&ct=g",
        "webp_size": "1365106",
        "webp": "https://media1.giphy.com/media/dSTZlrbH8u5jI6Hd51/giphy.webp?cid=34cf01dco5jtrtfj8psd7ykpeppfup8d6p8xy1qwssfsmise&ep=v1_gifs_trending&rid=giphy.webp&ct=g",
        "frames": "108",
        "hash": "f0f5bbd216242ac85fc75145d2dbb287"
    },
    "downsized": {
        "height": 480,
        "width": 480,
        "size": "1904302",
        "url": "https://media1.giphy.com/media/dSTZlrbH8u5jI6Hd51/giphy-downsized.gif?cid=34cf01dco5jtrtfj8psd7ykpeppfup8d6p8xy1qwssfsmise&ep=v1_gifs_trending&rid=giphy-downsized.gif&ct=g"
    },
    "downsized_large": {
        "height": 480,
        "width": 480,
        "size": "5513844",
        "url": "https://media1.giphy.com/media/dSTZlrbH8u5jI6Hd51/giphy.gif?cid=34cf01dco5jtrtfj8psd7ykpeppfup8d6p8xy1qwssfsmise&ep=v1_gifs_trending&rid=giphy.gif&ct=g"
    },
    "downsized_medium": {
        "height": 480,
        "width": 480,
        "size": "4141015",
        "url": "https://media1.giphy.com/media/dSTZlrbH8u5jI6Hd51/giphy-downsized-medium.gif?cid=34cf01dco5jtrtfj8psd7ykpeppfup8d6p8xy1qwssfsmise&ep=v1_gifs_trending&rid=giphy-downsized-medium.gif&ct=g"
    },
    "downsized_small": {
        "height": 250,
        "width": 250,
        "mp4_size": "117803",
        "mp4": "https://media1.giphy.com/media/dSTZlrbH8u5jI6Hd51/giphy-downsized-small.mp4?cid=34cf01dco5jtrtfj8psd7ykpeppfup8d6p8xy1qwssfsmise&ep=v1_gifs_trending&rid=giphy-downsized-small.mp4&ct=g"
    },
    "downsized_still": {
        "height": 480,
        "width": 480,
        "size": "38431",
        "url": "https://media1.giphy.com/media/dSTZlrbH8u5jI6Hd51/giphy-downsized_s.gif?cid=34cf01dco5jtrtfj8psd7ykpeppfup8d6p8xy1qwssfsmise&ep=v1_gifs_trending&rid=giphy-downsized_s.gif&ct=g"
    },
    "fixed_height": {
        "height": 200,
        "width": 200,
        "size": "991874",
        "url": "https://media1.giphy.com/media/dSTZlrbH8u5jI6Hd51/200.gif?cid=34cf01dco5jtrtfj8psd7ykpeppfup8d6p8xy1qwssfsmise&ep=v1_gifs_trending&rid=200.gif&ct=g",
        "mp4_size": "181984",
        "mp4": "https://media1.giphy.com/media/dSTZlrbH8u5jI6Hd51/200.mp4?cid=34cf01dco5jtrtfj8psd7ykpeppfup8d6p8xy1qwssfsmise&ep=v1_gifs_trending&rid=200.mp4&ct=g",
        "webp_size": "370024",
        "webp": "https://media1.giphy.com/media/dSTZlrbH8u5jI6Hd51/200.webp?cid=34cf01dco5jtrtfj8psd7ykpeppfup8d6p8xy1qwssfsmise&ep=v1_gifs_trending&rid=200.webp&ct=g"
    },
    "fixed_height_downsampled": {
        "height": 200,
        "width": 200,
        "size": "61701",
        "url": "https://media1.giphy.com/media/dSTZlrbH8u5jI6Hd51/200_d.gif?cid=34cf01dco5jtrtfj8psd7ykpeppfup8d6p8xy1qwssfsmise&ep=v1_gifs_trending&rid=200_d.gif&ct=g",
        "webp_size": "37136",
        "webp": "https://media1.giphy.com/media/dSTZlrbH8u5jI6Hd51/200_d.webp?cid=34cf01dco5jtrtfj8psd7ykpeppfup8d6p8xy1qwssfsmise&ep=v1_gifs_trending&rid=200_d.webp&ct=g"
    },
    "fixed_height_small": {
        "height": 100,
        "width": 100,
        "size": "317518",
        "url": "https://media1.giphy.com/media/dSTZlrbH8u5jI6Hd51/100.gif?cid=34cf01dco5jtrtfj8psd7ykpeppfup8d6p8xy1qwssfsmise&ep=v1_gifs_trending&rid=100.gif&ct=g",
        "mp4_size": "63882",
        "mp4": "https://media1.giphy.com/media/dSTZlrbH8u5jI6Hd51/100.mp4?cid=34cf01dco5jtrtfj8psd7ykpeppfup8d6p8xy1qwssfsmise&ep=v1_gifs_trending&rid=100.mp4&ct=g",
        "webp_size": "137944",
        "webp": "https://media1.giphy.com/media/dSTZlrbH8u5jI6Hd51/100.webp?cid=34cf01dco5jtrtfj8psd7ykpeppfup8d6p8xy1qwssfsmise&ep=v1_gifs_trending&rid=100.webp&ct=g"
    },
    "fixed_height_small_still": {
        "height": 100,
        "width": 100,
        "size": "3748",
        "url": "https://media1.giphy.com/media/dSTZlrbH8u5jI6Hd51/100_s.gif?cid=34cf01dco5jtrtfj8psd7ykpeppfup8d6p8xy1qwssfsmise&ep=v1_gifs_trending&rid=100_s.gif&ct=g"
    },
    "fixed_height_still": {
        "height": 200,
        "width": 200,
        "size": "11488",
        "url": "https://media1.giphy.com/media/dSTZlrbH8u5jI6Hd51/200_s.gif?cid=34cf01dco5jtrtfj8psd7ykpeppfup8d6p8xy1qwssfsmise&ep=v1_gifs_trending&rid=200_s.gif&ct=g"
    },
    "fixed_width": {
        "height": 200,
        "width": 200,
        "size": "991874",
        "url": "https://media1.giphy.com/media/dSTZlrbH8u5jI6Hd51/200w.gif?cid=34cf01dco5jtrtfj8psd7ykpeppfup8d6p8xy1qwssfsmise&ep=v1_gifs_trending&rid=200w.gif&ct=g",
        "mp4_size": "181984",
        "mp4": "https://media1.giphy.com/media/dSTZlrbH8u5jI6Hd51/200w.mp4?cid=34cf01dco5jtrtfj8psd7ykpeppfup8d6p8xy1qwssfsmise&ep=v1_gifs_trending&rid=200w.mp4&ct=g",
        "webp_size": "370024",
        "webp": "https://media1.giphy.com/media/dSTZlrbH8u5jI6Hd51/200w.webp?cid=34cf01dco5jtrtfj8psd7ykpeppfup8d6p8xy1qwssfsmise&ep=v1_gifs_trending&rid=200w.webp&ct=g"
    },
    "fixed_width_downsampled": {
        "height": 200,
        "width": 200,
        "size": "61701",
        "url": "https://media1.giphy.com/media/dSTZlrbH8u5jI6Hd51/200w_d.gif?cid=34cf01dco5jtrtfj8psd7ykpeppfup8d6p8xy1qwssfsmise&ep=v1_gifs_trending&rid=200w_d.gif&ct=g",
        "webp_size": "37136",
        "webp": "https://media1.giphy.com/media/dSTZlrbH8u5jI6Hd51/200w_d.webp?cid=34cf01dco5jtrtfj8psd7ykpeppfup8d6p8xy1qwssfsmise&ep=v1_gifs_trending&rid=200w_d.webp&ct=g"
    },
    "fixed_width_small": {
        "height": 100,
        "width": 100,
        "size": "317518",
        "url": "https://media1.giphy.com/media/dSTZlrbH8u5jI6Hd51/100w.gif?cid=34cf01dco5jtrtfj8psd7ykpeppfup8d6p8xy1qwssfsmise&ep=v1_gifs_trending&rid=100w.gif&ct=g",
        "mp4_size": "49485",
        "mp4": "https://media1.giphy.com/media/dSTZlrbH8u5jI6Hd51/100w.mp4?cid=34cf01dco5jtrtfj8psd7ykpeppfup8d6p8xy1qwssfsmise&ep=v1_gifs_trending&rid=100w.mp4&ct=g",
        "webp_size": "137944",
        "webp": "https://media1.giphy.com/media/dSTZlrbH8u5jI6Hd51/100w.webp?cid=34cf01dco5jtrtfj8psd7ykpeppfup8d6p8xy1qwssfsmise&ep=v1_gifs_trending&rid=100w.webp&ct=g"
    },
    "fixed_width_small_still": {
        "height": 100,
        "width": 100,
        "size": "3748",
        "url": "https://media1.giphy.com/media/dSTZlrbH8u5jI6Hd51/100w_s.gif?cid=34cf01dco5jtrtfj8psd7ykpeppfup8d6p8xy1qwssfsmise&ep=v1_gifs_trending&rid=100w_s.gif&ct=g"
    },
    "fixed_width_still": {
        "height": 200,
        "width": 200,
        "size": "11488",
        "url": "https://media1.giphy.com/media/dSTZlrbH8u5jI6Hd51/200w_s.gif?cid=34cf01dco5jtrtfj8psd7ykpeppfup8d6p8xy1qwssfsmise&ep=v1_gifs_trending&rid=200w_s.gif&ct=g"
    },
    "looping": {
        "mp4_size": "2506800",
        "mp4": "https://media1.giphy.com/media/dSTZlrbH8u5jI6Hd51/giphy-loop.mp4?cid=34cf01dco5jtrtfj8psd7ykpeppfup8d6p8xy1qwssfsmise&ep=v1_gifs_trending&rid=giphy-loop.mp4&ct=g",
        "width": null,
        "height": null
    },
    "original_still": {
        "height": 480,
        "width": 480,
        "size": "70510",
        "url": "https://media1.giphy.com/media/dSTZlrbH8u5jI6Hd51/giphy_s.gif?cid=34cf01dco5jtrtfj8psd7ykpeppfup8d6p8xy1qwssfsmise&ep=v1_gifs_trending&rid=giphy_s.gif&ct=g"
    },
    "original_mp4": {
        "height": 480,
        "width": 480,
        "mp4_size": "639713",
        "mp4": "https://media1.giphy.com/media/dSTZlrbH8u5jI6Hd51/giphy.mp4?cid=34cf01dco5jtrtfj8psd7ykpeppfup8d6p8xy1qwssfsmise&ep=v1_gifs_trending&rid=giphy.mp4&ct=g"
    },
    "preview": {
        "height": 298,
        "width": 298,
        "mp4_size": "35482",
        "mp4": "https://media1.giphy.com/media/dSTZlrbH8u5jI6Hd51/giphy-preview.mp4?cid=34cf01dco5jtrtfj8psd7ykpeppfup8d6p8xy1qwssfsmise&ep=v1_gifs_trending&rid=giphy-preview.mp4&ct=g"
    },
    "preview_gif": {
        "height": 105,
        "width": 105,
        "size": "47340",
        "url": "https://media1.giphy.com/media/dSTZlrbH8u5jI6Hd51/giphy-preview.gif?cid=34cf01dco5jtrtfj8psd7ykpeppfup8d6p8xy1qwssfsmise&ep=v1_gifs_trending&rid=giphy-preview.gif&ct=g"
    },
    "preview_webp": {
        "height": 168,
        "width": 168,
        "size": "40122",
        "url": "https://media1.giphy.com/media/dSTZlrbH8u5jI6Hd51/giphy-preview.webp?cid=34cf01dco5jtrtfj8psd7ykpeppfup8d6p8xy1qwssfsmise&ep=v1_gifs_trending&rid=giphy-preview.webp&ct=g"
    },
    "hd": {
        "height": 1000,
        "width": 1000,
        "mp4_size": "3730182",
        "mp4": "https://media1.giphy.com/media/dSTZlrbH8u5jI6Hd51/giphy-hd.mp4?cid=34cf01dco5jtrtfj8psd7ykpeppfup8d6p8xy1qwssfsmise&ep=v1_gifs_trending&rid=giphy-hd.mp4&ct=g"
    },
    "480w_still": {
        "height": 480,
        "width": 480,
        "size": "5513844",
        "url": "https://media1.giphy.com/media/dSTZlrbH8u5jI6Hd51/480w_s.jpg?cid=34cf01dco5jtrtfj8psd7ykpeppfup8d6p8xy1qwssfsmise&ep=v1_gifs_trending&rid=480w_s.jpg&ct=g"
    }
}
```